src/transform.js

```diff
@@ -21,9 +21,9 @@ export default function transform (node, transformations, o) {
 	return _transform(node, transformations, o);
 }
 
-function _transform (node, transformations, o = {}, property, parent) {
+function _transform (node, transformations, o = {}, property, parent, index) {
 	if (Array.isArray(node)) {
-		return node.map(n => _transform(n, transformations, o, property, parent));
+		return node.map((n, i) => _transform(n, transformations, o, property, parent, i));
```

Comment:
Actually, it looks like we may need *less* complexity: wouldn't we now be able to get all this info from the parent pointers? Previously we needed this because we did not store this information.

Reply:
That would rely on assuming parents are set, which is not a requirement of using transform

Reply:
We can call `setAll()` in `transform()`, remember there's no downside to this now :)